src/SprykerEco/Zed/Computop/Persistence/ComputopRepository.php

```diff
@@ -0,0 +1,116 @@
+<?php
+
+/**
+ * MIT License
+ * Use of this software requires acceptance of the Evaluation License Agreement. See LICENSE file.
+ */
+
+namespace SprykerEco\Zed\Computop\Persistence;
+
+use Generated\Shared\Transfer\ComputopPaymentComputopDetailTransfer;
+use Generated\Shared\Transfer\ComputopPaymentComputopOrderItemCollectionTransfer;
+use Generated\Shared\Transfer\ComputopPaymentComputopTransfer;
+use Generated\Shared\Transfer\ComputopSalesOrderItemCollectionTransfer;
+use Spryker\Zed\Kernel\Persistence\AbstractRepository;
+
+/**
+ * @method \SprykerEco\Zed\Computop\Persistence\ComputopPersistenceFactory getFactory();
+ */
+class ComputopRepository extends AbstractRepository implements ComputopRepositoryInterface
+{
+    /**
+     * @param string $transactionId
+     *
+     * @return \Generated\Shared\Transfer\ComputopPaymentComputopTransfer|null
+     */
+    public function findComputopPaymentByComputopTransId(string $transactionId): ?ComputopPaymentComputopTransfer
+    {
+        $computopPaymentEntity = $this->getFactory()
+            ->getPaymentComputopQuery()
+            ->filterByTransId($transactionId)
+            ->findOne();
+
+        if ($computopPaymentEntity === null) {
+            return null;
+        }
+
+        return $this->getFactory()
+            ->createComputopMapper()
+            ->mapComputopPaymentEntityToComputopPaymentTransfer($computopPaymentEntity, new ComputopPaymentComputopTransfer());
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\ComputopPaymentComputopTransfer $computopPaymentComputopTransfer
+     *
+     * @return \Generated\Shared\Transfer\ComputopSalesOrderItemCollectionTransfer
+     */
+    public function getComputopSalesOrderItemsCollection(
```

Comment:
```suggestion
    public function getComputopSalesOrderItemCollection(
```

as was mentioned before